### PR TITLE
Openldap provisioning : Reset phase implementation

### PR DESCRIPTION
**Pull request by shahidSTX**
# Problem Statement
- Implement reset phase for openldap provisioning

# Design
-  Reset phase details are implemented as per design 

# Coding 
- [x] Coding conventions are followed and code is consistent.
- [x] Confirm All CODACY errors are resolved.

# Testing 
- [x] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability.
- [x] Confirm Testing was performed with installed RPM. 

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [x] Is there a change in filename/package/module or signature [Y/N]: N
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [x] Check Acceptance Criterion is defined. 
- [x] All the tests performed should be mentioned before Resolving a JIRA. 
- [x] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

File: py-utils/src/utils/setup/openldap/resetcmd.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+import os
+from cortx.utils.log import Log
+from setupcmd import SetupCmd, OpenldapPROVError
+
+class ResetCmd(SetupCmd):
```
Comment (shalakadharap) [inline py-utils/src/utils/setup/openldap/resetcmd.py:25]: Why are we inheriting from setupcmd?

Comment (shahidSTX) [reply py-utils/src/utils/setup/openldap/resetcmd.py:25]: Not used immediately but in future if we are to use any definitions from SetupCmd which provides base, we need to include this.

Comment (shalakadharap) [reply py-utils/src/utils/setup/openldap/resetcmd.py:25]: @bkirunge7 
Can you comment here.
Will we be needing anything in RESET from setupcmd in future?
RESET only works on cleanup tasks

File: py-utils/src/utils/setup/openldap/templates/openldap.cleanup.tmpl.1-node [hunk 2]
```diff
@@ -0,0 +1 @@
+# No keys required for prepare
```
Comment (vimalraghwani20) [inline py-utils/src/utils/setup/openldap/templates/openldap.cleanup.tmpl.1-node:1]: it should be cleanup

File: py-utils/src/utils/setup/openldap/templates/openldap.reset.tmpl.1-node [hunk 3]
```diff
@@ -0,0 +1 @@
+# No keys required for prepare
```
Comment (vimalraghwani20) [inline py-utils/src/utils/setup/openldap/templates/openldap.reset.tmpl.1-node:1]: No keys required for reset

File: py-utils/src/utils/setup/openldap/resetcmd.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+    def delete_log_files(self):
+        """Delete log files."""
+        #delete log files
+        Log.debug("Starting log file deletion")
+        logFiles = ["/var/log/seagate/utils/openldap/OpenldapProvisioning",
+                    "/var/log/slapd.log"]
+        for logFile in logFiles:
+            self._delete_file(logFile)
+        Log.debug("Reset completed, empty log file")
```
Comment (vimalraghwani20) [inline py-utils/src/utils/setup/openldap/resetcmd.py:61]: empty log file??

File: py-utils/src/utils/setup/openldap/openldap.py [hunk 5]
```diff
@@ -266,5 +266,12 @@ def test(self, plan, config: str):
...
         """ Performs Configuraiton reset. Raises exception on error """
 
+        phase_name = "reset"
+        Log.debug("%s - Starting\n" % phase_name)
+        self.validate(phase_name)
+        self._keys_validate(phase_name) 
+        from resetcmd import ResetCmd
+        ResetCmd().process()
+        Log.debug("%s - Successful" % phase_name)
         # TODO: Perform actual steps. Obtain inputs using Conf.get(index, ..)
```
Comment (shahidSTX) [inline py-utils/src/utils/setup/openldap/openldap.py:276]: Remove this comment.

File: py-utils/src/utils/setup/openldap/openldap.py [hunk 6]
```diff
@@ -266,5 +266,12 @@ def test(self, plan, config: str):
     def reset(self):
         """ Performs Configuraiton reset. Raises exception on error """
```
Comment (shahidSTX) [inline py-utils/src/utils/setup/openldap/openldap.py:267]: Spell check

File: py-utils/src/utils/setup/openldap/resetcmd.py (hunk 4 above)
Comment (shahidSTX) [reply py-utils/src/utils/setup/openldap/resetcmd.py:61]: Yes, once reset is done, log is empty and hence we update this msg to it.
@vimalraghwani20 Do you have any suggestions for better msg?

File: py-utils/src/utils/setup/openldap/resetcmd.py (hunk 1 above)
Comment (vimalraghwani20) [reply py-utils/src/utils/setup/openldap/resetcmd.py:25]: @shalakadharap  @shahidSTX @bkirunge7  
I feel every single initialization which is required in multiple phases should go in setupcmd.
e.g.
Log.init('OpenldapProvisioning','/var/log/seagate/utils/openldap',\
             level='DEBUG')
why should we are initializing this in every phase ???

**Review by shalakadharap — APPROVED** [review]

**Review by vimalraghwani20 — APPROVED** [review]

File: py-utils/src/utils/setup/openldap/resetcmd.py (hunk 1 above)
Comment (shahidSTX) [reply py-utils/src/utils/setup/openldap/resetcmd.py:25]: Let us take this as a separate discussion. Closing this conversation here.

File: py-utils/src/utils/setup/openldap/templates/openldap.cleanup.tmpl.1-node (hunk 2 above)
Comment (shahidSTX) [reply py-utils/src/utils/setup/openldap/templates/openldap.cleanup.tmpl.1-node:1]: Done

File: py-utils/src/utils/setup/openldap/templates/openldap.reset.tmpl.1-node (hunk 3 above)
Comment (shahidSTX) [reply py-utils/src/utils/setup/openldap/templates/openldap.reset.tmpl.1-node:1]: Done.

File: py-utils/src/utils/setup/openldap/openldap.py (hunk 5 above)
Comment (shahidSTX) [reply py-utils/src/utils/setup/openldap/openldap.py:276]: Done.

File: py-utils/src/utils/setup/openldap/openldap.py (hunk 6 above)
Comment (shahidSTX) [reply py-utils/src/utils/setup/openldap/openldap.py:267]: Done.

File: py-utils/src/utils/setup/openldap/openldap.py [hunk 7]
```diff
@@ -264,7 +258,12 @@ def test(self, plan, config: str):
...
     def reset(self):
-        """ Performs Configuraiton reset. Raises exception on error """
-
-        # TODO: Perform actual steps. Obtain inputs using Conf.get(index, ..)
+        """ Performs Configuration reset. Raises exception on error """
+        phase_name = "reset"
+        Log.debug("%s - Starting\n" % phase_name)
+        self.validate(phase_name)
+        self._keys_validate(phase_name) 
+        from resetcmd import ResetCmd
```
Comment (abhilekh268) [inline py-utils/src/utils/setup/openldap/openldap.py:266]: can we keep this import up....with other imports?

File: py-utils/src/utils/setup/openldap/resetcmd.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+class ResetCmd(SetupCmd):
+    """Config Setup Cmd."""
+    Log.init('OpenldapProvisioning','/var/log/seagate/utils/openldap',\
+             level='DEBUG')
+    def __init__(self):
+        """Constructor"""
+        try:
+            super(ResetCmd, self).__init__()
+            Log.debug("Initializing reset phase")
```
Comment (abhilekh268) [inline py-utils/src/utils/setup/openldap/resetcmd.py:33]: have this log statement befor super call,that;s the point of initialization start.

File: py-utils/src/utils/setup/openldap/resetcmd.py [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+            try:
+                os.remove(filepath)
+            except Exception:
+                Log.debug("Failed deleting log file")
+
+    def delete_log_files(self):
+        """Delete log files."""
+        #delete log files
+        Log.debug("Starting log file deletion")
+        logFiles = ["/var/log/seagate/utils/openldap/OpenldapProvisioning",
```
Comment (abhilekh268) [inline py-utils/src/utils/setup/openldap/resetcmd.py:57]: can we read this log path from some config file?

File: py-utils/src/utils/setup/openldap/openldap.py (hunk 7 above)
Comment (shahidSTX) [reply py-utils/src/utils/setup/openldap/openldap.py:266]: Noted.

File: py-utils/src/utils/setup/openldap/resetcmd.py (hunk 8 above)
Comment (shahidSTX) [reply py-utils/src/utils/setup/openldap/resetcmd.py:33]: Noted.

File: py-utils/src/utils/setup/openldap/resetcmd.py (hunk 9 above)
Comment (shahidSTX) [reply py-utils/src/utils/setup/openldap/resetcmd.py:57]: Noted.

**Pull request merged**